src/main/java/edu/aau/groupc/canteenbackend/user/dto/UserDto.java

```diff
@@ -0,0 +1,59 @@
+package edu.aau.groupc.canteenbackend.user.dto;
+
+import edu.aau.groupc.canteenbackend.dto.DTO;
+import edu.aau.groupc.canteenbackend.user.User;
+import lombok.Data;
+import org.hibernate.validator.constraints.Length;
+
+import javax.validation.constraints.NotBlank;
+import javax.validation.constraints.Pattern;
+import java.io.Serializable;
+import java.util.Objects;
+
+@Data
+public class UserDto implements DTO, Serializable {
+
+    @NotBlank(message = "Username is required")
+    @Length(min = 3, message = "Username must be at least 3 characters long!")
+    @Length(max = 24, message = "Username cannot be longer than 24 characters!")
+    @Pattern(regexp = "^[A-Za-z0-9]+$", message = "Username can contain only alphanumeric characters!")
+    private String username;
+
+    @NotBlank(message = "Password is required")
+    @Pattern(regexp = "^[a-fA-F0-9]+$", message = "Password can contain only hexadecimal digits!")
+    @Length(min = 64, max = 64, message = "Password must be exactly 64 characters long!")
```

Comment:
Why have this arbitrary, very restricting password length requirement?
Consider relaxing the requirements, maybe a length of 8-64 characters or even completely remove the upper limit.

Reply:
> Nicely done, looks almost ready to merge to me.
> 
> However, I have a few small questions / suggestions still (also see comments below):
> 
> * Is there a way to query the type (i.e., `OWNER, ADMIN, USER`) of a user? This would be somewhat required for the frontend to know which widgets to display for a certain user.
>   Maybe just add a simple GET endpoint to the user controller which returns the user type given an auth token. This could be easily done using the `@Secured` annotation and `request.getAttribute("user").getType` in the endpoint method.
> * Maybe add one more commit addressing the [code smells identified by SonarCloud](https://sonarcloud.io/project/issues?id=elsantner_canteenMgmtBackend&pullRequest=31&resolved=false&types=CODE_SMELL). Should be very simple fixes but improves the code quality.

No idea how to reply to your original message (the one on the root level) so I used Quote.

This is a great idea. I completely forgot about this. Thanks!
Will definitely implement it. What I intend to do is to use the GET request for /user without any parameters to fetch these info.

Reply:
It is actually `/api/register` GET request which is `@Secured`.